src/platform/web/ThemeLoader.ts

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2020 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import type {Platform} from "./Platform.js";
+
+export enum COLOR_SCHEME_PREFERENCE { DARK, LIGHT, }
+
+export class ThemeLoader {
+    private _platform: Platform;
+    private _themeMapping: Record<string, string> = {};
+
+    constructor(platform: Platform) {
+        this._platform = platform;
+    }
+
+    async init(manifestLocations: string[]): Promise<void> {
+        for (const manifestLocation of manifestLocations) {
+            const { body } = await this._platform
+                .request(manifestLocation, {
+                    method: "GET",
+                    format: "json",
+                    cache: true,
+                })
+                .response();
+            Object.assign(this._themeMapping, body["source"]["built-asset"]);
```

Comment:
This takes some time to understand if you don't know off the top of your head what `build-asset` (also, shouldn't it be `built-assets`, plural?) contains... Could you write it out a bit or add a comment?

Reply:
> also, shouldn't it be built-assets, plural?

I can change this in another PR!